### PR TITLE
feat(golang): add table registry namespace

### PR DESCRIPTION
**Pull request by jplanckeel**
Description
Add table_scaleway_registry_namespacetable to scaleway plugin, allowing list and get on registry namespace.

I need a person to test this table ! Else i test this in ~2-3 weeks..

# Example query results

```sql
select
  name,
  id,
  status,
  created_at,
  region,
  project,
  organization
from
  scaleway_registry_namespace
where
  is_public = true;
```

<details>
  <summary>Results</summary>
```
> select
  name,
  id,
  status,
  created_at,
  region,
  project,
  organization
from
  scaleway_registry_namespace
where
  is_public = true;
+---------------------+--------------------------------------+--------+---------------------------+--------+---------+--------------+
| name                | id                                   | status | created_at                | region | project | organization |
+---------------------+--------------------------------------+--------+---------------------------+--------+---------+--------------+
| namespace-amsterdam | 6cf7ef0c-f950-4f5e-a3b4-c757954e645b | ready  | 2022-12-08T10:22:10+01:00 | nl-ams | <null>  | <null>       |
+---------------------+--------------------------------------+--------+---------------------------+--------+---------+--------------+
```
</details>

### CODE REVIEW

**Comment by misraved** [comment]
Thanks @jplanckeel  for adding this new table👍.  Great work !!!

From an initial overview, the changes look pretty solid, however, I feel we should add 1-2 more examples highlighting the important features of the new table 👍.

I will look to review this by the end of this week so that we can push for a release 😃 .

File: scaleway/plugin.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"scaleway_object_bucket":           tableScalewayObjectBucket(ctx),
 			"scaleway_rdb_database":            tableScalewayRDBDatabase(ctx),
 			"scaleway_rdb_instance":            tableScalewayRDBInstance(ctx),
+			"scaleway_registry_namespace":		tableScalewayRegistryNamespace(ctx),
```
Comment (misraved) [inline scaleway/plugin.go:33]:
```suggestion
			"scaleway_registry_namespace":	tableScalewayRegistryNamespace(ctx),
```

File: scaleway/table_scaleway_registry_namespace.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+			},
+			{
+				Name:        "is_public",
+				Description: "The namespace visibility policy.",
+				Type:        proto.ColumnType_BOOL,
+				Transform:   transform.FromField("IsPublic").Transform(transform.ToBool),
+			},
+			{
+				Name:        "size",
+				Description: "total size of the namespace, calculated as the sum of the size of all images in the namespace.",
```
Comment (misraved) [inline scaleway/table_scaleway_registry_namespace.go:74]:
```suggestion
				Description: "Total size of the namespace, calculated as the sum of the size of all images in the namespace.",
```

File: scaleway/table_scaleway_registry_namespace.go [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+			},
+			{
+				Name:        "image_count",
+				Description: "The number of images in the namespace.",
+				Type:        proto.ColumnType_INT,
+				Transform:   transform.FromField("ImageCount").Transform(transform.ToInt),
+			},
+			{
+				Name:        "region",
+				Description: "Specifies the region where the namespace.",
```
Comment (misraved) [inline scaleway/table_scaleway_registry_namespace.go:96]:
```suggestion
				Description: "Specifies the region where the namespace is located.",
```

File: scaleway/table_scaleway_registry_namespace.go [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+	parseRegionData, err := scw.ParseRegion(region)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_regsitry_namespace.listRegsitryNamespace", "region_parsing_error", err)
+		return nil, err
+	}
+
+	quals := d.KeyColumnQuals
+	if quals["region"] != nil && quals["region"].GetStringValue() != region {
+		return nil, nil
+	}
```
Comment (misraved) [inline scaleway/table_scaleway_registry_namespace.go:138]:
```suggestion
	if quals["region"] == nil && quals["region"].GetStringValue() == region {
		return nil, nil
	}
```
Shouldn't it be returning nil when the quals are nil?

**Review by misraved — COMMENTED** [review]
Please take a look at the review comments. Thanks!!

Please note that haven't tested the table with the resources yet, hoping to get it done soon 👍 .

File: scaleway/table_scaleway_registry_namespace.go (hunk 4 above)
Comment (jplanckeel) [reply scaleway/table_scaleway_registry_namespace.go:138]: maybe, i read the code. for me we don't list regsitry if we have not a region in the list and if the region on connection isn't the same.. 

i have adapted this code https://github.com/turbot/steampipe-plugin-scaleway/blob/main/scaleway/table_scaleway_instance_server.go#L225  to change zone by region.

**Comment by jplanckeel** [comment]
> Thanks @jplanckeel for adding this new table👍. Great work !!!
> 
> From an initial overview, the changes look pretty solid, however, I feel we should add 1-2 more examples highlighting the important features of the new table 👍.
> 
> I will look to review this by the end of this week so that we can push for a release 😃 .

Thank you, i hope test this PR next week :)

**Comment by jplanckeel** [comment]
Hi, i have tested this table, it's work :)  forme is good and i have update docs

File: docs/tables/scaleway_registry_namespace.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  status,
+  created_at,
+  region,
+  project,
+  organization
+from
+  scaleway_registry_namespace;
+```
+
+### List public registry namespace
```
Comment (cbruno10) [inline docs/tables/scaleway_registry_namespace.md:22]:
```suggestion
### List public registry namespaces
```

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@jplanckeel I've added a few suggestions and questions, can you please have a look when you get a chance? Thanks!

File: scaleway/table_scaleway_registry_namespace.go [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+				Name:        "is_public",
+				Description: "The namespace visibility policy.",
+				Type:        proto.ColumnType_BOOL,
+				Transform:   transform.FromField("IsPublic").Transform(transform.ToBool),
+			},
+			{
+				Name:        "size",
+				Description: "Total size of the namespace, calculated as the sum of the size of all images in the namespace.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Size").Transform(transform.ToString),
```
[the remark this answers is not in the document]
Comment (jplanckeel) [reply scaleway/table_scaleway_registry_namespace.go]: Is fixed and tested ;)

File: scaleway/table_scaleway_registry_namespace.go [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listRegistryNamespace(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
```
[the remark this answers is not in the document]
Comment (jplanckeel) [reply scaleway/table_scaleway_registry_namespace.go]: i fixed all typo Registry !

File: scaleway/table_scaleway_registry_namespace.go [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+		}
+		req.Page = scw.Int32Ptr(*req.Page + 1)
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getRegsitryNamespace(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
[the remark this answers is not in the document]
Comment (jplanckeel) [reply scaleway/table_scaleway_registry_namespace.go]: i fixed all typo Registry !

**Review by cbruno10 — APPROVED** [review]

**Comment by cbruno10** [comment]
Thanks @jplanckeel for the contribution!

**Comment by jplanckeel** [comment]
Thanks you for the review :)

**Pull request merged**